executor/show.go

```diff
@@ -691,33 +691,38 @@ func (e *ShowExec) fetchShowCreateTable() error {
 	partitionInfo := tb.Meta().Partition
 	if partitionInfo != nil {
 		// this if statement takes care of range columns case
-		if partitionInfo.Columns != nil && partitionInfo.Type == model.PartitionTypeRange {
-			buf.WriteString(fmt.Sprintf("\nPARTITION BY RANGE COLUMNS("))
-			for i, col := range partitionInfo.Columns {
-				buf.WriteString(col.L)
-				if i < len(partitionInfo.Columns)-1 {
-					buf.WriteString(",")
-				}
-			}
-			buf.WriteString(") (\n")
+		if partitionInfo.Type == model.PartitionTypeHash {
+			buf.WriteString(fmt.Sprintf("\nPARTITION BY HASH( %s )", partitionInfo.Expr))
```

Comment:
you can use `fmt.Fprintf(buf, ("", xx))`

Reply:
@zz-jason done.